### PR TITLE
simplify code for reading tags

### PR DESCRIPTION
**Pull request by funniray**
also:
- Transformed seekHead object into a new object to make reading tags easier
- Throws error if file doesn't have a seekHead

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -106,180 +110,43 @@ export class Metadata {
...
   const Segment = await metadata.getSegment()
   const SeekHead = await metadata.getSeekHead()
-  const Info = await metadata.getInfo()
-  const Tracks = await metadata.getTracks()
-  const Chapters = await metadata.getChapters()
-  const Clusters = await metadata.getClusters()
-  const Cues = await metadata.getCues()
-  const Attachments = await metadata.getAttachments()
-  const Tags = await metadata.getTags()
+  const Info = await metadata.readSeekedTag("Info")
```
Comment (ThaUnknown) [inline index.js]: single quotes not double

File: index.js [hunk 2]
```diff
@@ -53,37 +53,41 @@ const readChildren = (parent = {}, start = 0) => {
...
     this.segmentStart = null
     this.seekHead = null
-    this.info = null
-    this.tracks = []
-    this.chapters = []
-    this.clusters = []
-    this.cues = []
-    this.attachments = []
-    this.tags = []
+    this.Info = []
```
Comment (ThaUnknown) [inline index.js]: no.... keep them lowercase

File: index.js [hunk 3]
```diff
@@ -53,37 +53,41 @@ const readChildren = (parent = {}, start = 0) => {
...
+  if (!seekHeadTags) throw new Error('Couldn\'t find seek head');
   const seekHead = readChildren(seekHeadTags)
   // Determines if there is a second SeekHead referenced by the first SeekHead.
   // Note: If true, the first must *only* contains a reference to the second, so no other tags will be in the first.
-  const recursive = seekHead.Children.find((child) => {
-    return child.Seek.SeekID === EbmlTagId.SeekHead
+
+  const transformedHead = {}
+
+  seekHead.Children.forEach(child => {
```
Comment (ThaUnknown) [inline index.js]: since you're here, use for const of

File: index.js [hunk 4]
```diff
@@ -106,180 +110,43 @@ export class Metadata {
...
-      if (!seekHeadTags) return Promise.resolve([])
+    if (this[tag].length === 0) {
+      if (!this.seekHead[tag]) return Promise.resolve([])
 
-      const tagsStream = this.blob.slice(this.segmentStart + seekHeadTags.Seek.SeekPosition).stream()
-      return readUntilTag(tagsStream, EbmlTagId.Tags, true).then((tags) => {
-        this.tags = readChildren(tags, this.segmentStart + seekHeadTags.Seek.SeekPosition)
-        return this.tags
+      const stream = this.blob.slice(this.segmentStart + this.seekHead[tag]).stream()
+      return readUntilTag(stream, EbmlTagId[tag], true).then((child) => {
```
Comment (ThaUnknown) [inline index.js]: since you're here revamp this to use await instead of .then

File: index.js [hunk 5]
```diff
@@ -106,180 +110,43 @@ export class Metadata {
...
-
-    if (this.tags.length === 0) {
-      const seekHeadTags = this.seekHead.Children.find((seek) => {
-        const seekId = parseInt(seek.Seek.SeekID.toString('hex'), 16)
-        return seekId === EbmlTagId.Tags
-      })
-
-      if (!seekHeadTags) return Promise.resolve([])
+    if (this[tag].length === 0) {
+      if (!this.seekHead[tag]) return Promise.resolve([])
```
Comment (ThaUnknown) [inline index.js]: return []

File: index.js [hunk 6]
```diff
@@ -50,25 +51,26 @@ const readChildren = (parent = {}, start = 0) => {
...
+  for (const child of seekHead.Children) {
+    if (!child.Seek) return // CRC32 elements will appear, currently we don't check them
+    const tagName = EbmlTagId[Tools.readUnsigned(child.Seek.SeekID)]
+    transformedHead[tagName] = child.Seek.SeekPosition
+  }
 
   if (transformedHead.SeekHead) {
-    const seekHeadStream = this.blob.slice(transformedHead.SeekHead + segmentStart).stream();
+    const seekHeadStream = createStream(transformedHead.SeekHead + segmentStart)
     return readSeekHead(seekHeadStream, segmentStart)
```
Comment (ThaUnknown) [inline index.js]: createStream is not passed here, also overall dont like the funcntion

Comment (funniray) [reply index.js]: It's passed through from getSeekHead on line 114. Don't want to access this.blob from outside of the class. Could move `readSeekHead` inside of the class or just pass through the blob to the function

Comment (ThaUnknown) [reply index.js]: its not passed to readSeekHead on L74

Comment (funniray) [reply index.js]: From my understanding, you shouldn't have more than 1 recursive SeekHead, but might as well just in case

**Comment by funniray** [comment]
I don't know how to implement this, but if the first non-crc32 element inside of the Segment isn't a SeekHead element, then we should assume that there's no SeekHead in the file and not try to read the entire file looking for a SeekHead

**Pull request merged**